lib/livebook_web/live/output/input_component.ex

```diff
@@ -305,6 +367,35 @@ defmodule LivebookWeb.Output.InputComponent do
     {:ok, html_value}
   end
 
+  defp parse(html_value, %{type: :datetime}) do
+    case NaiveDateTime.from_iso8601(html_value) do
+      {:ok, datetime} -> {:ok, datetime}
+      {:error, _error} -> :error
+    end
+  end
+
+  defp parse(html_value, %{type: :time}) do
+    case Time.from_iso8601(html_value) do
+      {:ok, time} -> {:ok, time}
+      {:error, _error} -> :error
+    end
+  end
+
+  defp parse(html_value, %{type: :date}) do
+    case Date.from_iso8601(html_value) do
+      {:ok, date} -> {:ok, date}
+      {:error, _error} -> :error
+    end
+  end
+
+  defp parse(html_value, %{type: :week}) do
+    {:ok, html_value}
+  end
+
+  defp parse(html_value, %{type: :month}) do
+    {:ok, html_value}
+  end
```

Comment:
We parse values into more structured data whenever possible, so I think in this case we should parse them into tuples `{year, week}` and `{year, month}` respectively.